Automattic/wp-calypso
# Replace Lynx patterns with virtual themes

Fixes https://github.com/Automattic/wp-calypso/issues/75289
Requires D107928-code
Requires https://github.com/Automattic/wp-calypso/pull/75782

## Proposed Changes

Refactor the design/pattern picker logic of the Link in Bio tailored flow so it reuses the virtual themes functionality: 

- Fetch designs from the `starter-designs` endpoint (effectively moving away from the `ptk/patterns` endpoint).
- Use `applyThemeWithPatterns` action to apply the design that has been assembled in the virtual theme.
- Remove logic that is no longer used.

This will allow us to retire the Lynx theme, so designers only need to maintain Link in Bio patterns in the `dotcompatterns` site.

## Testing Instructions

- Use the Calypso live link below.
- Go to `/setup/link-in-bio`.
- Proceed to the design/patter picker step.
- Explore all designs and ensure they match the designs that are currently in production (*).
  - (*) Note: You should find 3 more designs now that are not in production because there are more Link in Bio virtual themes than patterns in `lynxdemo` (see paYJgx-3eb-p2).
- Choose a design and complete the flow.
- Make sure that Blank Canvas has been activated (instead of Lynx).
- Make sure that the site has a homepage template containing a pattern (instead of a page containing a pattern displayed by the blank template).
